azurerm/resource_arm_network_watcher_flow_log_test.go

```diff
@@ -0,0 +1,560 @@
+		if err != nil {
+			return err
+		}
+
+		resourceGroupName := parsedID.ResourceGroup
+		networkWatcherName := parsedID.Path["networkWatchers"]
+
+		client := testAccProvider.Meta().(*ArmClient).watcherClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+		statusParameters := network.FlowLogStatusParameters{
+			TargetResourceID: &networkSecurityGroupID,
+		}
+		future, err := client.GetFlowLogStatus(ctx, resourceGroupName, networkWatcherName, statusParameters)
+		if err != nil {
+			return fmt.Errorf("Error retrieving Flow Log Configuration for target %q (Network Watcher %q / Resource Group %q): %+v", networkSecurityGroupID, networkWatcherName, resourceGroupName, err)
+		}
+
+		err = future.WaitForCompletionRef(ctx, client.Client)
+		if err != nil {
+			return fmt.Errorf("Error waiting for retrieval of Flow Log Configuration for target %q (Network Watcher %q / Resource Group %q): %+v", networkSecurityGroupID, networkWatcherName, resourceGroupName, err)
+		}
+
+		_, err = future.Result(client)
+		if err != nil {
+			return fmt.Errorf("Error retrieving of Flow Log Configuration for target %q (Network Watcher %q / Resource Group %q): %+v", networkSecurityGroupID, networkWatcherName, resourceGroupName, err)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMNetworkWatcherFlowLog_basicConfig(rInt int, rString string, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+    name     = "acctestRG-%d"
+    location = "%s"
+}
+
+resource "azurerm_network_security_group" "test" {
+    name                = "acctestnsg%d"
+    location            = "${azurerm_resource_group.test.location}"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_network_watcher" "test" {
+    name                = "acctestnw-%d"
+    location            = "${azurerm_resource_group.test.location}"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_storage_account" "test" {
+    name                = "acctestsa%s"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    location            = "${azurerm_resource_group.test.location}"
+
+    account_tier              = "Standard"
+    account_replication_type  = "LRS"
+    enable_https_traffic_only = true
+}
+
+resource "azurerm_network_watcher_flow_log" "test" {
+    network_watcher_name = "${azurerm_network_watcher.test.name}"
+    resource_group_name  = "${azurerm_resource_group.test.name}"
+
+    network_security_group_id = "${azurerm_network_security_group.test.id}"
+    storage_account_id        = "${azurerm_storage_account.test.id}"
+    enabled                   = true
+
+    retention_policy {
+        days    = 0
+        enabled = false
+    }
+}
+`, rInt, location, rInt, rInt, rString)
+}
+
+func testAccAzureRMNetworkWatcherFlowLog_retentionPolicyConfig(rInt int, rString string, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
```

Comment:
Not a blocker, but we could [use a template for the resources we depend on](https://github.com/terraform-providers/terraform-provider-azurerm/blob/e6e7fc6a8107a4edabb2ddd38135bea9f790aee0/azurerm/resource_arm_automation_schedule_test.go#L338) so shorten these tests